src/http/Message.h

```diff
@@ -136,6 +140,11 @@ class Message : public RefCountable
     virtual bool parseFirstLine(const char *blk_start, const char *blk_end) = 0;
 
     virtual void hdrCacheInit();
+    /// Parses headers stored in a buffer, when the header size is unknown.
+    /// \returns 1 and sets hdr_sz on success
+    /// \returns 0 when needs more data
+    /// \returns -1 on error
+    virtual int parseHeaderUnknownLength(const char *buf, const size_t bufLen, const bool atEnd, size_t &hdrLen) = 0;
```

Comment:
Please see if you can replace these two new virtual parseHeader*Length() methods with a single virtual configureContentLengthInterpreter(interpreter) method that the old parsing code can call to configure the interpreter (with the status code for responses).

Reply:
Done.